server/webapp/WEB-INF/rails/webpack/views/components/modal/index.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2018 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {MithrilComponent} from "../../../jsx/mithril-component";
+import * as m from 'mithril';
+import * as _ from 'lodash';
+import * as styles from './index.scss';
+import {ModalManager} from "./modal_manager";
+import * as Buttons from "../buttons";
+
+const uuid4 = require('uuid/v4');
+
+export abstract class Modal extends MithrilComponent<any> {
+  public id: string;
+
+  protected constructor() {
+    super();
+    this.id = `modal-${uuid4()}`;
+  }
+
+  abstract title(): string;
+
+  abstract body(): JSX.Element;
+
+  buttons(): JSX.Element[] {
+    return [<Buttons.Primary onclick={this.close.bind(this)}>OK</Buttons.Primary>];
+  }
+
+  render() {
+    ModalManager.render(this);
+  }
+
+  close() {
+    ModalManager.close(this);
+  }
+
+  handleEscapeKey() {
+    $('body').on(`keydown.${this.id}`, (e) => {
```

Comment:
Can we consider weaning off jquery? This may mean holding a reference to the function, which we can later uninstall ourselves?